spack/packages/2decomp-fft/package.py

```diff
@@ -20,15 +20,46 @@ class _2decompFft(MakefilePackage):
 
     version('1.5.847', sha256='b137d7cf9b771de0a174d1e6c4ff0e48b4a84b51ff6c62140b4f522092e6784f')
 
+    variant(
+        'backend', default='generic', description='FFT backend',
```

Comment:
I doubt `generic` is a backend. Some code has to compute locally either `fftw3` or `mkl`. Does `2decomp` ship with reference implementation of FFT?

Reply:
All I know is what their documentation says.  From Makefile.inc.x86:

```
# Choose one FFT engine, available options are:
#   acml       - AMD Core Math Library
#   ffte       - FFTE
#   fftpack5   - FFTPACK5
#   fftw3      - FFTW version 3.x
#   fftw3_f03  - FFTW 3.3-beta1 or later (with Fortran 2003 interface)
#   generic    - A general FFT algorithm (no 3rd-party library needed)
#   mkl        - Intel Math Kernel Library
```

Reply:
Both `generic` and `fftw3` backends build for me for `2decom-fft` package